pkg/utils/kubernetes.go

```diff
@@ -66,3 +67,40 @@ func PodIP(pod corev1.Pod) (string, error) {
 
 	return "", fmt.Errorf("pod %s/%s does not have an IP address", pod.Namespace, pod.Name)
 }
+
+// PodNames return the name of the pods in a list
+func PodNames(pods []corev1.Pod) []string {
+	names := []string{}
+	for _, pod := range pods {
+		names = append(names, pod.Name)
+	}
```

Comment:
We can probably preallocate the `names` slice here. It's not like it's the hot path, but maybe just as a "best practices" approach:
```suggestion
	names := make([]string, 0, len(pods))
	for _, pod := range pods {
		names = append(names, pod.Name)
	}
```